src/analysis/complex/basic.lean

```diff
@@ -267,6 +265,16 @@ lemma isometry_of_real : isometry (coe : ℝ → ℂ) := of_real_li.isometry
 
 @[continuity] lemma continuous_of_real : continuous (coe : ℝ → ℂ) := of_real_li.continuous
 
+/-- The only continuous ring homomorphism from `ℝ` to `ℂ` is the identity. -/
+lemma ring_hom_eq_of_real_of_continuous {f : ℝ →+* ℂ} (h : continuous f) :
+  f = complex.of_real :=
+begin
+  refine fun_like.coe_injective _,
+  refine rat.dense_embedding_coe_real.dense.equalizer h continuous_of_real _,
+  ext1,
+  simp only [function.comp_app, map_rat_cast, of_real_rat_cast],
+end
```

Comment:
I'm surprised we don't have the first lemma:
```suggestion
lemma _root_.alg_hom.eq_of_id {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
  (f : R →ₐ[R] A) : f = algebra.of_id R A :=
by { ext, apply alg_hom.commutes }

/-- The only continuous ring homomorphism from `ℝ` to `ℂ` is the identity. -/
lemma ring_hom_eq_of_real_of_continuous {f : ℝ →+* ℂ} (h : continuous f) :
  f = complex.of_real :=
by convert congr_arg alg_hom.to_ring_hom (alg_hom.mk' f $ map_real_smul f h).eq_of_id; ext1; refl
```

Reply:
Oh but we do have [alg_hom.subsingleton](https://leanprover-community.github.io/mathlib_docs/algebra/algebra/subalgebra/basic.html#alg_hom.subsingleton), so the proof could also be
```
begin
  convert congr_arg alg_hom.to_ring_hom
    (subsingleton.elim (alg_hom.mk' f $ map_real_smul f h) $ algebra.of_id ℝ ℂ),
  ext1, refl,
end 
```
without introducing the first lemma.

Reply:
Oh whoops, I didn't mean to commit this lemma at all.